sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -523,22 +523,62 @@ pi_result piMemBufferCreate(pi_context context, pi_mem_flags flags, size_t size,
                             void *host_ptr, pi_mem *ret_mem,
                             const pi_mem_properties *properties) {
   pi_result ret_err = PI_INVALID_OPERATION;
-  clCreateBufferWithPropertiesINTEL_fn FuncPtr = nullptr;
-
-  if (properties)
+  if (properties) {
+    clCreateBufferWithPropertiesINTEL_fn FuncPtr = nullptr;
+    const size_t propSize = sizeof(properties) / sizeof(pi_mem_properties);
     // First we need to look up the function pointer
     ret_err = getExtFuncFromContext<clCreateBufferWithPropertiesName,
                                     clCreateBufferWithPropertiesINTEL_fn>(
         context, &FuncPtr);
+    if (FuncPtr) {
+      std::vector<pi_mem_properties> supported(properties,
+                                               properties + propSize);
+      // Go through buffer properties. If there is one, that shall be propagated
+      // to an OpenCL runtime - check if this property is being supported.
+      for (auto prop = supported.begin(); prop != supported.end(); ++prop) {
+        // Check if PI_MEM_CHANNEL_INTEL property is supported. If it's not -
+        // just ignore it, as it's an optimization hint.
+        if (*prop == PI_MEM_CHANNEL_INTEL) {
+          size_t deviceCount;
+          cl_int ret_err =
+              clGetContextInfo(cast<cl_context>(context), CL_CONTEXT_DEVICES, 0,
+                               nullptr, &deviceCount);
+          if (ret_err != CL_SUCCESS || deviceCount < 1)
+            return PI_INVALID_CONTEXT;
+          std::vector<cl_device_id> devicesInCtx(deviceCount);
+          ret_err = clGetContextInfo(
+              cast<cl_context>(context), CL_CONTEXT_DEVICES,
+              deviceCount * sizeof(cl_device_id), devicesInCtx.data(), nullptr);
+
+          size_t retSize;
+          ret_err = clGetDeviceInfo(devicesInCtx[0], CL_DEVICE_EXTENSIONS, 0,
+                                    nullptr, &retSize);
+          if (ret_err != CL_SUCCESS)
+            return PI_INVALID_DEVICE;
+          std::string extensions(retSize, '\0');
+          ret_err = clGetDeviceInfo(devicesInCtx[0], CL_DEVICE_EXTENSIONS,
+                                    retSize, &extensions[0], nullptr);
+          if (ret_err != CL_SUCCESS)
+            return PI_INVALID_DEVICE;
+
+          size_t pos = extensions.find("cl_intel_mem_channel_property");
+          if (pos == std::string::npos)
+            supported.erase(prop);
+        }
+      }
+      if (!supported.empty()) {
```

Comment:
Don't you want to assert for unsupported properties rather than sending them blindly to OpenCL R?T

Reply:
Unsupported properties aren't being sent to OpenCL RT, see lines 564-566. Also, IMHO assertions show an assertion shows an assumption that a programmer make in the code. I don't assume, that every property shall be supported by a low level RT, instead I want these properties be ignored as they are optimization hints (so far PI_MEM_PROPERTIES_CHANNEL is the only one, that can be passed to a lower runtime, but I think, that for the upcoming properties mechanism will be the same).

Reply:
Done, thanks!